python/pyiceberg/table/__init__.py

```diff
@@ -527,8 +519,13 @@ def history(self) -> List[SnapshotLogEntry]:
         """Get the snapshot history of this table."""
         return self.metadata.snapshot_log
 
-    def update_schema(self) -> UpdateSchema:
-        return UpdateSchema(self.schema(), self)
+    def update_schema(self, allow_incompatible_changes: bool = False, case_sensitive: bool = True) -> UpdateSchema:
+        return UpdateSchema(self, allow_incompatible_changes=allow_incompatible_changes, case_sensitive=case_sensitive)
+
+    def _do_commit(self, request: CommitTableRequest) -> None:
```

Comment:
I think it would make more sense to pass requirements and updates. The identifier is already in the table and this can create the request. Or maybe the request should be created in `catalog._commit_table`?

Reply:
Yes, I think that's cleaner as well, updated!